api/client.md

```diff
@@ -0,0 +1,646 @@
+```go
+// Client http methods
+func (c *Client) Get(url string) *Agent
+func (c *Client) Head(url string) *Agent
+func (c *Client) Post(url string) *Agent
+func (c *Client) Put(url string) *Agent
+func (c *Client) Patch(url string) *Agent
+func (c *Client) Delete(url string) *Agent
+```
+{% endcode %}
+
+## ✨ Agent
+`Agent` is on top of [fasthttp `HostClient`](https://github.com/valyala/fasthttp/blob/master/client.go#L603) which has lots of convenient helper methods and several `send request` methods.
+
+### Parse
+Parse initializes URI and HostClient.
+
+{% code title="Parse" %}
+```go
+a := AcquireAgent()
+req := a.Request()
+req.Header.SetMethod(MethodGet)
+req.SetRequestURI("http://example.com")
+
+if err := a.Parse(); err != nil {
+	panic(err)
+}
+
+code, body, errs := a.Bytes() // ...
+```
+{% endcode %}
+
+### Set
+Set sets the given 'key: value' header.
+
+{% code title="Signature" %}
+```go
+func (a *Agent) Set(k, v string) *Agent
+func (a *Agent) SetBytesK(k []byte, v string) *Agent
+func (a *Agent) SetBytesV(k string, v []byte) *Agent
+func (a *Agent) SetBytesKV(k []byte, v []byte) *Agent
+```
+{% endcode %}
+
+{% code title="Example" %}
+```go
+
+agent.Set("k1", "v1").
+    SetBytesK([]byte("k1"), "v1").
+    SetBytesV("k1", []byte("v1")).
+    SetBytesKV([]byte("k2"), []byte("v2"))
+// ...
+```
+{% endcode %}
+
+### Add
+Add adds the given 'key: value' header. Multiple headers with the same key may be added with this function.
+
+{% code title="Signature" %}
+```go
+func (a *Agent) Add(k, v string) *Agent
+func (a *Agent) AddBytesK(k []byte, v string) *Agent
+func (a *Agent) AddBytesV(k string, v []byte) *Agent
+func (a *Agent) AddBytesKV(k []byte, v []byte) *Agent
+```
+{% endcode %}
+
+{% code title="Example" %}
+```go
+
+agent.Add("k1", "v1").
+    AddBytesK([]byte("k1"), "v1").
+    AddBytesV("k1", []byte("v1")).
+    AddBytesKV([]byte("k2"), []byte("v2"))
+// ...
+```
+{% endcode %}
+
+### ConnectionClose
+ConnectionClose sets 'Connection: close' header.
+
+{% code title="Signature" %}
+```go
+func (a *Agent) ConnectionClose() *Agent
+```
+{% endcode %}
+
+{% code title="Example" %}
+```go
+agent.ConnectionClose()
+// ...
+```
+{% endcode %}
+
+### UserAgent
+UserAgent sets User-Agent header value.
+
+{% code title="Signature" %}
+```go
+func (a *Agent) UserAgent(userAgent string) *Agent
+func (a *Agent) UserAgentBytes(userAgent []byte) *Agent
+```
+{% endcode %}
+
+{% code title="Example" %}
+```go
+agent.UserAgent("fiber")
+// ...
+```
+{% endcode %}
+
+### Cookie
+Cookie sets 'key: value' cookie.
+
+{% code title="Signature" %}
+```go
+func (a *Agent) Cookie(key, value string) *Agent
+func (a *Agent) CookieBytesK(key []byte, value string) *Agent
+func (a *Agent) CookieBytesKV(key, value []byte) *Agent
+func (a *Agent) Cookies(kv ...string) *Agent
+func (a *Agent) CookiesBytesKV(kv ...[]byte) *Agent
+```
+{% endcode %}
+
+{% code title="Example" %}
+```go
+agent.Cookie("k", "v")
+agent.Cookies("k1", "v1", "k2", "v2")
```

Comment:
Are we able use a map too?

```
m := map[string]string{
  "k1": "v1",
  "k2": "v2",
}
agent.Cookies(m)
```

Reply:
Yes we can. But if we do so, we have to declare a map struct.